### ISSUE TITLE
Unable to create new stream from an existing stream with camel cased field

### ISSUE THREAD

**Issue by tiffanycitra** [confluentinc/ksql#5032]
**Describe the bug**
I created a stream with a camel cased field and am unable to create a new stream from it when I have a `WHERE` clause involving the camel cased field.

**To Reproduce**
Version of KSQL is 5.4.0.
```
// create test topic
kafka-topics --zookeeper localhost:2181 --create --topic test_topic --partitions 4 --replication-factor 1 

// launch ksql-cli
ksql

 // create a stream that consumes from the test topic
ksql> create stream `test-stream` (`camelCasedField` VARCHAR) with (kafka_topic='test_topic', value_format='json');

// create a new stream from the first stream
ksql> create stream `test-stream-2` as select * from `test-stream` where `camelCasedField` = 'foo';
io.confluent.ksql.util.KsqlStatementException: Column 'test-stream.CAMELCASEDFIELD' cannot be resolved.
```

**Expected behavior**
`test-stream-2` is created successfully.

**Actual behaviour**
See above.

**Additional context**
I tried other variations with the back tick and they seem to be interpolated correctly:
```
ksql> create stream `test-stream-2` as select * from `test-stream` where `camelCased_Field` = 'foo';
Column 'test-stream.camelCased_Field' cannot be resolved.
ksql> create stream `test-stream-2` as select * from `test-stream` where `camelCasedField->blah` = 'foo';
Column 'test-stream.camelCasedField->blah' cannot be resolved.
ksql> create stream `test-stream-2` as select * from `test-stream` where `nonexisting_underscore_field` = 'foo';
Column 'test-stream.nonexisting_underscore_field' cannot be resolved.
```

**Comment by apurvam** [comment]
@agavra  was preserving case for field names part of the quoted identifier work you did? If so, did it land in 5.4?

**Comment by stevenpyzhang** [comment]
Sorry, I should've closed this out after I cherry-picked the fix into 5.4.

This isn't an issue in master
